### PR TITLE
Unused Code

### PR DESCRIPTION
**Pull request by SoriUR**

### CODE REVIEW

File: xcode/UnusedConfig.txt [hunk 1]
```diff
@@ -0,0 +1,5 @@
+--ignored-resources:
```
Comment (petropavel13) [inline xcode/UnusedConfig.txt]: use yaml format?

File: xcode/build_phases/Unused.rb [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+  def parse_arguments()
+      resources = []
+
+      arguments = ARGV.clone
+
+      until arguments.empty?
+          item = arguments.shift
+
+          # find --config file
+          if item == $config_argv_key
```
Comment (petropavel13) [inline xcode/build_phases/Unused.rb]: use https://docs.ruby-lang.org/en/2.1.0/OptionParser.html instead?

**Review by petropavel13 — CHANGES_REQUESTED** [review]

File: xcode/UnusedConfig.yml [hunk 3]
```diff
@@ -0,0 +1,5 @@
+--ignored-resources:
```
Comment (petropavel13) [inline xcode/UnusedConfig.yml]: `--` is legal in yaml?

File: xcode/build_phases/Unused.rb [hunk 4]
```diff
@@ -0,0 +1,188 @@
...
+            opts.on("-i", "--ignore=a, b, c", Array) { |i| options[:ignore] += i }
+
+        end.parse!
+
+        # find --config file
+        if !options[:config].nil?
+            fileName = options[:config]
+            resources += YAML.load_file(fileName).fetch("--ignored-resources")
+            elsif
+            puts "---------\n Warning: Config file is now provided \n---------"
```
Comment (petropavel13) [inline xcode/build_phases/Unused.rb]: not?

File: xcode/UnusedConfig.yml (hunk 3 above)
Comment (petropavel13) [reply xcode/UnusedConfig.yml]: do like swiftlint:
https://gist.github.com/tweetjay/1b7f00b6f312a5a2cf7b5676420b9b0c#file-swiftlint-yml-L3

**Review by petropavel13 — CHANGES_REQUESTED** [review]

**Review by petropavel13 — DISMISSED** [review]

**Review by petropavel13 — DISMISSED** [review]

**Review by Viveron — APPROVED** [review]

**Review by petropavel13 — APPROVED** [review]

**Pull request merged**